toolchain/check/check_unit.cpp

```diff
@@ -395,8 +395,11 @@ auto CheckUnit::CheckRequiredDefinitions() -> void {
         break;
       }
       case CARBON_KIND(SemIR::ImplDecl impl_decl): {
-        if (!context_.impls().Get(impl_decl.impl_id).is_defined()) {
-          emitter_.Emit(decl_inst_id, MissingDefinitionInImpl);
+        auto& impl = context_.impls().Get(impl_decl.impl_id);
+        if (!impl.is_defined()) {
+          CARBON_DIAGNOSTIC(MissingImplDefinition, Error,
+                            "no definition found for declaration of impl");
```

Comment:
Maybe "impl declared but not defined"? I think we have previously said that we prefer diagnostics that state facts about the program rather than actions of the toolchain (eg, say the thing doesn't exist, not that we couldn't find it).

Reply:
Done.